src/data/fintype/basic.lean

```diff
@@ -1636,3 +1636,25 @@ variables [fintype α] [decidable_eq α]
 count_eq_one_of_mem finset.univ.nodup (finset.mem_univ _)
 
 end multiset
+
+namespace fintype
+
+/-- An induction principle for finite types, analogous to `nat.rec`. It effectively says
+that every `fintype` is either `empty` or `option α`, up to an `equiv`. -/
+noncomputable def induction_empty_option {P : Type u → Sort v}
```

Comment:
I'm hesitant about this landing in `Sort`, unless there are lemmas describing the output. This definition constructs data (which might not be subsingleton) and so we should be able to talk about properties of the resulting data; for instance if I use this definition on an empty type, how does the result relate to `h_empty`? I think you should either try to come up with lemmas to make this definition usable, or move it back to Prop (in which case these issues go away)

Reply:
I'm also not really sure what the purpose of a recursor landing in `Sort` is if it's non-computable - you can already lift a Prop value to Sort using choice.

I expect the computable version of this definition would conclude `trunc (P a)`

Reply:
Sure, let's just use `Prop`. There is not a lot you can say about the output, and to get a computable version with `trunc (P a)` we'd need another assumption about decidable equality anyway.